### DISCUSSION TITLE
How to incorporate two different directions of the drift velocity of a particular species

### DISCUSSION THREAD

**Discussion by deSUBHAM** [BLAST-WarpX/warpx#4376] [Simulation Inputs]
Hi,
I have a pair plasma, where the electrons have the Maxwell-Juttner momentum distribution with the drift velocities along x and z directions. The positrons also have the same momentum distribution but with drift velocities opposite to that of the electrons. I am using `beta_distribution_type`, `beta_function(x,y,z)`, and `bulk_vel_dir` to describe the drift velocity of each species. However, I am not sure how to incorporate two directions of the drift velocity. If I write the input file (for one of the species, say electrons) in the following way
```
electrons.momentum_distribution_type = "maxwell_juttner"
electrons.theta_distribution_type = "constant"
electrons.theta = temp
electrons.beta_distribution_type = "parser"
electrons.beta_function(x,y,z) = "-(sqrt(sigma)*de/Del)*(1/cosh(z/Del))*tanh(z/Del)" , "-(sqrt(sigma)*de/Del)*(1/cosh(z/Del))*(1/cosh(z/Del))"
electrons.bulk_vel_dir = "x" ,"y"
```
it throws error indicating `electrons.beta_function(x,y,z)`. My query is how to incorporate two directions of the drift velocities in the input file.
Thank you.

**Comment by palvib** [comment]
It would be great if someone can throw light upon how to include multiple components of beta.. right now it seems as though one is allowed to specify only a single component/single direction for the bulk velocity.

**Comment by hklion** [comment]
Currently it's only possibly to specify bulk velocity along a single direction. Adding multiple directions would be a useful feature, but will require some bookkeeping that I haven't gotten a chance to look at yet. The last few weeks have been busy with conferences, but I'll be working on this once I return.

**Comment by archermarx** [comment]
It would be a bit annoying, but if you use the PICMI python input, you can specify a `ParticleListDistribution` and manually load in particle velocities as you want. You'd need to write your own python code for sampling from a Maxwell-Juttner distribution, however.
